### PR TITLE
Minimal Python container to build `az aro` extension

### PR DESCRIPTION
**Pull request by tsatam**
(This PR recreates #3439 off of in-repository branch for CI checks)

### Which issue this PR addresses:

Small step for [ARO-5088](https://issues.redhat.com/browse/ARO-5088)

### What this PR does / why we need it:

- Creates a new Makefile target to build the `az aro` extension with no other dependencies
- Adds a new containerfile to create a build container for the above

### Test plan for issue:

Manually invoke target and ensure it creates an `aro-${VERSION}-py2.py3-none-any.whl` file within `./python/az/aro/dist`. 

### Is there any documentation that needs to be updated for this PR?

Eventually, https://msazure.visualstudio.com/AzureRedHatOpenShift/_wiki/wikis/ARO.wiki/188187/AzureCLI-preview-extension-publishing# should be updated to include this target. 

### Additional Notes
- This container does not perform the verification steps performed by the `azdev` command (`style`/`linter`/`test`). `azdev` requires that it be run within the context of a Git repository, as well as from within a specific virtualenv.

### CODE REVIEW

**Review by SudoBrendan — APPROVED** [review]

**Review by cadenmarchese — DISMISSED** [review]
makes sense 👍

File: Makefile [hunk 1]
```diff
@@ -274,4 +277,4 @@ vendor:
 install-go-tools:
 	go install ${GOTESTSUM}
 
-.PHONY: admin.kubeconfig aks.kubeconfig aro az ci-portal clean client deploy dev-config.yaml discoverycache generate image-aro-multistage image-fluentbit image-proxy init-contrib lint-go runlocal-rp proxy publish-image-aro-multistage publish-image-fluentbit publish-image-proxy secrets secrets-update e2e.test tunnel test-e2e test-go test-python vendor build-all validate-go unit-test-go coverage-go validate-fips install-go-tools
\ No newline at end of file
+.PHONY: admin.kubeconfig aks.kubeconfig aro az azext_aro ci-portal ci-rp clean client deploy dev-config.yaml discoverycache fix-macos-vendor generate image-aro-multistage image-fluentbit image-proxy init-contrib lint-go runlocal-rp proxy publish-image-aro-multistage publish-image-fluentbit publish-image-proxy secrets secrets-update e2e.test tunnel test-e2e test-go test-python vendor build-all validate-go unit-test-go coverage-go validate-fips install-go-tools
```
Comment (SudoBrendan) [inline Makefile:280]:
```suggestion
.PHONY: admin.kubeconfig aks.kubeconfig aro az azext-aro ci-portal ci-rp clean client deploy dev-config.yaml discoverycache fix-macos-vendor generate image-aro-multistage image-fluentbit image-proxy init-contrib lint-go runlocal-rp proxy publish-image-aro-multistage publish-image-fluentbit publish-image-proxy secrets secrets-update e2e.test tunnel test-e2e test-go test-python vendor build-all validate-go unit-test-go coverage-go validate-fips install-go-tools
```

**Review by SudoBrendan — CHANGES_REQUESTED** [review]
Just a quick typo fix. Otherwise, builds fine!

**Review by SudoBrendan — APPROVED** [review]
LGTM

**Review by SudoBrendan — APPROVED** [review]

**Review by tony-schndr — APPROVED** [review]
LGTM

**Review by shubhadapaithankar — APPROVED** [review]
LGTM

**Pull request merged**